lib/Doctrine/ORM/EntityManager.php

```diff
@@ -191,6 +200,16 @@ public function getMetadataFactory()
     }
 
     /**
+     * Gets the persister factory used to manipulate collection and entity classes.
+     *
+     * @return \Doctrine\ORM\Persisters\PersisterFactory
+     */
+    public function getPersisterFactory()
```

Comment:
TL;DR: the UnitOfWork as-is is already poop: let's stash poop in there instead of shoveling it on other classes.

That makes our EntityManager some kind of locator (meh).
I'd prefer to make a tradeoff here, and do this crappy thing in the UnitOfWork instead (yes, with all the deficiencies that come from it).

The UnitOfWork as we know it will go away at some point in time, while the EntityManager will stay and be consumed by userland.